bids/layout/layout.py

```diff
@@ -113,8 +113,23 @@ def __init__(self, root, validate=True, index_associated=True,
 
         # Validate arguments
         if not isinstance(root, six.string_types):
-            raise ValueError("root argument must be a string specifying the"
-                             " directory containing the BIDS dataset.")
+            # attempt to handle pathlib paths before giving up
+            try:
+                import pathlib
+                assert isinstance(root, pathlib.Path)
+                # the os functions can only handle pathlib.Paths from 3.4 when it joined
+                # the standard library
+                if not six.PY34:
+                    raise TypeError("root argument is pathlib.Path-derived, "
+                            "this type is only supported on python>=3.4")
```

Comment:
I thought there are backports

Reply:
pathlib itself is available in 2.7 *but* os.path only works with pathlib.Path instances from 3.4 onward.